symengine/functions.cpp

```diff
@@ -1181,7 +1182,7 @@ RCP<const Basic> trig_to_sqrt(const RCP<const Basic> &arg)
 }
 
 /* ---------------------------- */
-ASin::ASin(const RCP<const Basic> &arg) : TrigFunction(arg)
+ASin::ASin(const RCP<const Basic> &arg) : InverseTrigFunction(arg)
```

Comment:
There are other places in the code base where `TrigFunction`  is used for all types. Please fix those as well.